File: src/dataset/cpp/beer_song_Exercism_johnngugi/openAI/ChatGPT4_beer_song_v4.cpp
#include <string>
#include "beer_song.h"

namespace beer {

    const std::string one_bottle = "1 bottle of beer on the wall, 1 bottle of beer.\n"
                                   "Take it down and pass it around, no more bottles of beer on the wall.\n";

    const std::string zero_bottles = "No more bottles of beer on the wall, no more bottles of beer.\n"
                                     "Go to the store and buy some more, 99 bottles of beer on the wall.\n";

    std::string verse(int verse_num) {
        if (verse_num == 0) {
            return zero_bottles;
        }
        if (verse_num == 1) {
            return one_bottle;
        }

        return std::to_string(verse_num) + " bottles of beer on the wall, " + std::to_string(verse_num) +
               " bottles of beer.\nTake one down and pass it around, " +
               std::to_string(verse_num - 1) + ((verse_num - 1 == 1) ? 
               " bottle of beer on the wall.\n" : " bottles of beer on the wall.\n");
    }

    std::string sing(int from) {
        std::string result;
        for (int i = from; i >= 0; --i) {
            result += verse(i) + (i > 0 ? "\n" : "");
        }
        return result;
    }

    std::string sing(int from, int to) {
        std::string result;
        for (int i = from; i >= to; --i) {
            result += verse(i) + (i > to ? "\n" : "");
        }
        return result;
    }

}

File: src/dataset/typescript/beer-song_Exercism-typescript-FilipeCerejo/openAI/ChatGPT4_beer-song_v4.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function bottles(n) {
    if (n === 0)
        return 'no more bottles';
    if (n === 1)
        return '1 bottle';
    return `${n} bottles`;
}
function action(n) {
    return n === 0
        ? 'Go to the store and buy some more'
        : n === 1
            ? 'Take it down and pass it around'
            : 'Take one down and pass it around';
}
function capitalize(s) {
    return s.charAt(0).toUpperCase() + s.slice(1);
}
function verse(beerNumber) {
    const nextNumber = beerNumber === 0 ? 99 : beerNumber - 1;
    return `${capitalize(bottles(beerNumber))} of beer on the wall, ${bottles(beerNumber)} of beer.\n` +
        `${action(beerNumber)}, ${bottles(nextNumber)} of beer on the wall.\n`;
}
function sing(beerMax = 99, beerMin = 0) {
    let result = '';
    for (let i = beerMax; i >= beerMin; i--) {
        result += verse(i) + (i > beerMin ? '\n' : '');
    }
    return result;
}
exports.default = { verse, sing };
